docs/blog/2018-10-26-introducing-gatsby-themes/index.md

```diff
@@ -0,0 +1,154 @@
+history. The community around gatsby has built a lot of different
+starters for various use cases including blogging, working with
+material design, and documentation.
+
+The problem with starters is that they're one-offs. Starters are
+boilerplate projects that begin to diverge immediately from upstream
+and have no easy way of updating when changes are made
+upstream. There's another approach to boilerplate that has become
+popular in recent years that fixes some problems with the boilerplate
+approach such as updating with upstream. One such project is
+create-react-app. In the gatsby world, we can improve on starters
+similarly with themes.
+
+## Themes
+
+If a single gatsby-config encodes the functionality of a whole Gatsby
+site then if we can compose the gatsby-config data structure together
+we have the base for themes. We can encode portions of our
+gatsby-config as themes and re-use them across sites.
+
+# Why Themes?
+
+Defining themes as the composition of Gatsby sites allows us to start
+solving a variety of use cases. For example, when a site gets built as
+part of a wider product offering it's often the case that one team
+will build out a set of functionality, including branding elements,
+and the other teams will mostly consume this functionality. Themes
+allow us to distribute this functionality as an npm package and allow
+the customization of various branding elements through our
+gatsby-config.
+
+# Mechanics of Theming
+
+At a base level, theming simply combines the gatsby-config of the
+theme with the gatsby-config of your site. Since it's an experimental
+feature, we use an experimental namespace to declare themes in
+gatsby-config.
+
+```js
+module.exports = {
+  __experimentalThemes: [["gatsby-theme-blog", {}]],
+}
+```
+
+Themes often need to be parameterized for various reasons, such as
+changing the base url for subsections of a site or applying branding
+variables. We can do this through the theme options if we define our
+theme's gatsby-config as a function that returns an object.
+
+```js
+module.exports = {
+  __experimentalThemes: [
+    [
+      "gatsby-theme-blog",
+      {
+        some: "value",
+      },
+    ],
+  ],
+}
+```
+
+Themes also function as plugins and any config passed into the theme
+in your gatsby-config will also be passed to your theme's gatsby-\*
+files as plugin options. This allows themes to override any settings
+inherited from the theme's own plugin declarations or apply gatsby
+lifecycle hooks such as on-create-page.
+
+# Next Steps
+
+## Sub Themes and Overriding
+
+This is just the first step and it enables us to experiment with
+further improvements in userland before merging them into
+core. Sub-theming, for example, is a critical part of a theming
+ecosystem that is currently missing from core. Overriding theme
+elements is possible on a coarse level right now in userland. If, for
+example, a theme defines a set of pages using createPage we can define
```

Comment:
```suggestion
example, a theme defines a set of pages using [`createPage`](https://www.gatsbyjs.org/docs/actions/#createPage) we can define
```

Reply:
@ChristopherBiscardi I edited this to not use the domain. That's my fault, sorry.